src/MangaJet/app/src/main/java/com/mangajet/mangajet/mangareader/MangaReaderVPAdapter.kt

```diff
@@ -0,0 +1,122 @@
+package com.mangajet.mangajet.mangareader
+
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.davemorrissey.labs.subscaleview.ImageSource
+import com.davemorrissey.labs.subscaleview.SubsamplingScaleImageView
+import com.mangajet.mangajet.R
+import com.mangajet.mangajet.data.MangaPage
+
+class MangaReaderVPAdapter(viewModel: MangaReaderViewModel) :
+    RecyclerView.Adapter<MangaReaderVPAdapter.MangaReaderPageHolder>() {
+    companion object {
+        const val SINGLE_CHAPTER_REVERSED_LIST = 0
+        const val SIDE_CHAPTER_REVERSED_LIST   = 1
+        const val MIDDLE_CHAPTER_REVERSED_LIST = 2
+    }
+
+    // viewModel, which contains our interesting data
+    var currentViewModelWithData = viewModel
+
+    inner class MangaReaderPageHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
+        private val imagePage = itemView.findViewById<SubsamplingScaleImageView>(R.id.mangaPage)
+        private val pageAndChapterNumberView =
+            itemView.findViewById<TextView>(R.id.pageAndChapterNumber)
+
+        fun bind(mangaPage : MangaPage, chapter : Int, page : Int) {
+            val chapterOutput = chapter + 1
+            val pageOutput = page + 1
+
+            //pageAndChapterNumberView.text = "Chap: $chapterOutput / Page: $pageOutput"
```

Comment:
it better be in format
"Ch." + chapterOutput ". p." + pageOutput + "/" + totalPages